File: e-learning-service/els-graphql/src/main/java/org/example/elsgraphql/config/GraphqlConfig.java
package org.example.elsgraphql.config;

import graphql.scalars.ExtendedScalars;
import org.example.elsgraphql.directive.AuthenticationDirective;
import org.example.elsgraphql.directive.AuthorizationDirective;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.graphql.execution.RuntimeWiringConfigurer;

@Configuration
public class GraphqlConfig {

    @Bean
    public RuntimeWiringConfigurer runtimeWiringConfigurer(
            AuthenticationDirective authenticationDirective,
            AuthorizationDirective authorizationDirective
    ) {
        return wiringBuilder -> wiringBuilder
                .scalar(ExtendedScalars.Date)
                .scalar(ExtendedScalars.DateTime)
                .scalar(ExtendedScalars.GraphQLLong)
                .directive("authenticate", authenticationDirective)
                .directive("authorize", authorizationDirective)
                ;
    }
}

File: api-protocol/graphql/goopang/src/main/kotlin/org/example/goopang/config/GraphQLConfig.kt
package org.example.goopang.config

import graphql.scalars.ExtendedScalars
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.graphql.execution.RuntimeWiringConfigurer

@Configuration
class GraphQLConfig {

    @Bean
    fun runtimeWiringConfiguration(): RuntimeWiringConfigurer {
        return RuntimeWiringConfigurer { it.scalar(ExtendedScalars.DateTime) }
    }
}